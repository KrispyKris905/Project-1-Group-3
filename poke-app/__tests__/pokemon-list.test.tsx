import React from "react";
import { render, waitFor } from '@testing-library/react-native';
import ListScreen from "../app/pokemon-list";

test("renders the ListScreen component correctly", async () => {
  const { getByText } = render(<ListScreen />);

  // Wait for the Pokémon list to be fetched and rendered
  await waitFor(() => {
    expect(getByText("Pokémon List")).toBeTruthy();
  });

  await waitFor(() => {
    expect(getByText("bulbasaur")).toBeTruthy();
  });

});