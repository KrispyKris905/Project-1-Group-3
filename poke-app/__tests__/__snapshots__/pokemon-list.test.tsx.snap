// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListScreen component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "marginBottom": 20,
      }
    }
  >
    Pok√©mon List
  </Text>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
